pyrules/rules.py

```diff
@@ -1,93 +1,161 @@
+        if name:
+            self.name = name
+        self._current_ruleid = None
+
     def perform(self, context):
         count = 0
         for rule in self.rules:
-            if all([eval(condition, context.as_dict) for condition in rule['if']]):
+            evaluator = LogicEvaluator(
+                rule['if'].get('logic'), rule['if']['conditions'])
+            if evaluator.evaluate(context):
                 count = count + 1
-                self._current_ruleid = rule.get('id', count)
+                self._current_ruleid = rule.get('rule') or count
                 for action, target in zip(rule['then'], rule['target']):
-                    if context._translations:
-                        action = context._translations.replace(action)
-                        target = context._translations.replace(target)
-                    result = context[target.replace('context.', '').strip()] = eval(action, context.as_dict)
+                    #if context._translations:
+                    #    action = context._translations.replace(action)
+                    #    target = context._translations.replace(target)
+                    result = \
+                        context[target.replace('context.', '').strip()] = (
+                            eval(action, context.as_dict)
+                            if isinstance(action, basestring)
+                            else action)
                     self.record(context, result)
             else:
-                break
+                continue
         else:
             self._current_ruleid = None
             return True
         return False
+
     @property
     def ruleid(self):
         if self._current_ruleid:
-            return "%s.%s" % (super(TableRuleset, self).ruleid, self._current_ruleid)
-        return super(TableRuleset, self).ruleid
-    
+            return "%s.%s" % (super(TableRule, self).ruleid, self._current_ruleid)
+        return super(TableRule, self).ruleid
+
+    @classmethod
+    def from_yaml(cls, text):
+        return cls._from_data(yaml.load(text))
+        
+    @classmethod
+    def from_json(cls, text):
+        return cls._from_data(json.loads(text))
+
+    @classmethod
+    def _from_data(cls, data):
+        # We have to convert non-string data in clauses back to strings,
+        # because they will be eval-ed
+        rules = []
+        for rule in data['rules']:
+            obj = {
+                'rule': rule.get('rule'),
+                'then': rule['then'],
+                'target': rule['target']}
+            if_clause = {}
+            # Convert conditions to dictionaries, i.e. "foo" becomes {"foo": True}
+            conditions = rule['if'].get('conditions', [])
+            # Get logic string. If it's not specified, generate string like
+            # "1 & 2 & .. & N", where N is number of conditions.
+            logic = rule['if'].get('logic')
+            obj['if'] = {'logic': logic, 'conditions': conditions}
```

Comment:
I'll add a separate issue for this, since it wasn't requested in issue. The parser won't be complicated, but neither trivial, currently operators that are supported are:
*  bool comparisons
* attr lookup
* dict lookup
* string operators (contains/icontains)